entities/liquidity_provision.go

```diff
@@ -154,3 +157,38 @@ func (lp LiquidityProvision) ToRow() []interface{} {
 		lp.CommitmentAmount, lp.Fee, lp.Sells, lp.Buys, lp.Version,
 		lp.Status, lp.Reference, lp.VegaTime}
 }
+
+func (lp LiquidityProvision) Cursor() *Cursor {
+	lc := LiquidityProvisionCursor{
+		VegaTime: lp.VegaTime,
+		ID:       lp.ID.String(),
+	}
+	return NewCursor(lc.String())
+}
+
+func (lp LiquidityProvision) ToProtoEdge(_ ...any) *v2.LiquidityProvisionsEdge {
+	return &v2.LiquidityProvisionsEdge{
+		Node:   lp.ToProto(),
+		Cursor: lp.Cursor().Encode(),
+	}
+}
+
+type LiquidityProvisionCursor struct {
+	VegaTime time.Time `json:"vegaTime"`
+	ID       string    `json:"id"`
+}
+
+func (lc LiquidityProvisionCursor) String() string {
+	bs, err := json.Marshal(lc)
```

Comment:
I guess we do it elsewhere but I don't think the output of the json.Marshal function is guaranteed. Not sure if it's best here?

Reply:
ah doesn't matter actually for some reason, I though you were hashing it or something. but it's just marshalled in json...